src/EditorFeatures/Core.Wpf/Suggestions/FixAll/FixAllCodeRefactoringGetFixesService.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Internal.Log;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
+{
+    [ExportWorkspaceServiceFactory(typeof(IFixAllCodeRefactoringGetFixesService), ServiceLayer.Host), Shared]
+    internal class FixAllCodeRefactoringGetFixesService : IFixAllCodeRefactoringGetFixesService, IWorkspaceServiceFactory
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FixAllCodeRefactoringGetFixesService()
+        {
+        }
+
+        public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
+            => this;
+
+        public async Task<Solution?> GetFixAllChangedSolutionAsync(FixAllContext fixAllContext)
+        {
+            var codeAction = await GetFixAllCodeActionAsync(fixAllContext).ConfigureAwait(false);
+            if (codeAction == null)
+            {
+                return fixAllContext.Project.Solution;
```

Comment:
why is this returning .Project.Solution.  Code in prior class returned just .Solution.

Reply:
I thought I removed Solution property from this context because it was just returning Project.Solution. Let me clean this up using the Generics/Type hierarchy suggestion you provided and make it consistent.